SoftLayer/CLI/hardware/upgrade.py

```diff
@@ -57,7 +61,92 @@ def cli(env, identifier, memory, network, drive_controller, public_bandwidth, ad
             disks = {'description': 'resize_disk', 'capacity': guest_disk[0], 'number': guest_disk[1]}
             disk_list.append(disks)
 
-    if not mgr.upgrade(hw_id, memory=memory, nic_speed=network, drive_controller=drive_controller,
-                       public_bandwidth=public_bandwidth, disk=disk_list, test=test):
-        raise exceptions.CLIAbort('Hardware Server Upgrade Failed')
-    env.fout('Successfully Upgraded.')
+    response = mgr.upgrade(hw_id, memory=memory, nic_speed=network, drive_controller=drive_controller,
+                           public_bandwidth=public_bandwidth, disk=disk_list, test=test)
+
+    if response:
+        if test:
+            add_data_to_table(response, table)
+        else:
+            table.add_row(['order_date', response.get('orderDate')])
+            table.add_row(['order_id', response.get('orderId')])
+            add_data_to_table(response['orderDetails'], table)
+            place_order_table = get_place_order_information(response)
+            table.add_row(['Place Order Information', place_order_table])
```

Comment:
could by table.add_row(['place_order_information', place_order_table])

Reply:
Generally I'd prefer a more formal name for these sort of tables. the row `order_id` is ok since it directly ties to an API value, but even then "Order ID" would likely look better to end users.

Reply:
Done.